build-support/bin/native/bootstrap.sh

```diff
@@ -1,202 +0,0 @@
-  # We sincerely hope that no one ever runs `rustup override set` in a subdirectory of the working directory.
-  "${CARGO_HOME}/bin/rustup" override set "${RUST_TOOLCHAIN}" >&2
-  "${CARGO_HOME}/bin/rustup" component add rustfmt-preview >&2
-  "${CARGO_HOME}/bin/rustup" component add rust-src >&2
-
-  # Sometimes fetching a large git repo dependency can take more than 10 minutes.
-  # This times out on travis, because nothing is printed to stdout/stderr in that time.
-  # Pre-fetch those git repos and keep writing to stdout as we do.
-  # Noisily wait for a cargo fetch. This is not using run_cargo so that the process which is
-  # killed by _wait_noisily on ^C is the cargo process itself, rather than a rustup wrapper which
-  # spawns a separate cargo process.
-  # We need to do this before we install grpcio-compiler to pre-fetch its repo.
-  _wait_noisily "${CARGO_HOME}/bin/cargo" fetch --manifest-path "${NATIVE_ROOT}/Cargo.toml" || die
-
-  if [[ ! -x "${CARGO_HOME}/bin/cargo-ensure-installed" ]]; then
-    "${CARGO_HOME}/bin/cargo" install cargo-ensure-installed >&2
-  fi
-  "${CARGO_HOME}/bin/cargo" ensure-installed --package=cargo-ensure-installed --version=0.1.0 >&2
-  "${CARGO_HOME}/bin/cargo" ensure-installed --package=protobuf --version=1.4.2 >&2
-  "${CARGO_HOME}/bin/cargo" ensure-installed --package=grpcio-compiler --version=0.2.0 >&2
-
-  local download_binary="${REPO_ROOT}/build-support/bin/download_binary.sh"
-  local -r cmakeroot="$("${download_binary}" "binaries.pantsbuild.org" "cmake" "3.9.5" "cmake.tar.gz")" || die "Failed to fetch cmake"
-  local -r goroot="$("${download_binary}" "binaries.pantsbuild.org" "go" "1.7.3" "go.tar.gz")/go" || die "Failed to fetch go"
-
-  export GOROOT="${goroot}"
-  export EXTRA_PATH_FOR_CARGO="${cmakeroot}/bin:${goroot}/bin"
-}
-
-function run_cargo() {
-  # Exports $EXTRA_PATH_FOR_CARGO which should be put on the $PATH
-  ensure_native_build_prerequisites || die
-
-  if [[ "${ensure_cffi_sources}" == "1" ]]; then
-    # Must happen in the pants venv and have PANTS_SRCPATH set.
-    _ensure_cffi_sources || die
-  fi
-
-  local -r cargo="${CARGO_HOME}/bin/cargo"
-  # We change to the ${REPO_ROOT} because if we're not in a subdirectory of it, .cargo/config isn't picked up.
-  (cd "${REPO_ROOT}" && PATH="${EXTRA_PATH_FOR_CARGO}:${PATH}" "${cargo}" "$@")
-}
-
-function _wait_noisily() {
-  "$@" &
-  pid=$!
-  trap 'kill ${pid} ; exit 130' SIGINT
-
-  i=0
-  while ps -p "${pid}" >/dev/null 2>/dev/null; do
-    [[ "$((i % 60))" -eq 0 ]] && echo >&2 "[Waiting for $@ (pid ${pid}) to complete]"
-    i="$((i + 1))"
-    sleep 1
-  done
-
-  wait "${pid}"
-
-  trap - SIGINT
-}
-
-function _build_native_code() {
-  # Builds the native code, and echos the path of the built binary.
-
-  ensure_cffi_sources=1 run_cargo build ${MODE_FLAG} --manifest-path ${NATIVE_ROOT}/Cargo.toml -p engine || die
-  echo "${NATIVE_ROOT}/target/${MODE}/libengine.${LIB_EXTENSION}"
-}
-
-function bootstrap_native_code() {
```

Comment:
This broke out into `bootstrap_code.sh` which uses the broken-out `bootstrap_rust.sh`.